doc/client_lua_api : clarify how client side api+a mod starts (#8822)
minetest/minetest: doc/client_lua_api.txt

Mods are loaded during client startup from the mod load paths by running
the `init.lua` scripts in a shared environment.

> In order to load client-side mods in your worlds, you have 2 conditions :

Reviewer: IMO, it's bad practice to use pronouns like "you" in documentation. Using passive voice might be a better solution here.

```suggestion
In order to load client-side mods in a world, the following conditions need to be satisfied:
```

>  1) $path_user/minetest.conf contains the setting `enable_client_modding = true`

Reviewer: ```suggestion
1) $path_user/minetest.conf contains the setting `enable_client_modding = true`
```

Removing this space will correctly render this as an ordered list

> 2) The client-side mod located in $path_user/clientmods/<modname> is added to

Reviewer: `<modname>` doesn't render correctly, and I suspect that this is because it is interpreted as a HTML tag. Enclosing all filepaths in backticks (like in [lua_api.txt](https://github.com/minetest/minetest/blob/master/doc/lua_api.txt)) fixes this.

>  2) The client-side mod located in $path_user/clientmods/<modname> is added to

Reviewer: ```suggestion
2) The client-side mod located in $path_user/clientmods/<modname> is added to
```

Same as above

>     $path_user/clientmods/mods.conf as `load_mod_<modname> = true` in order to load*.

Reviewer 2: The final `in order to load` is redundant, due to `In order to load client-side mods in your worlds, you have 2 conditions :`

>     $path_user/clientmods/mods.conf as `load_mod_<modname> = true` *.

Reviewer: ```suggestion
    $path_user/clientmods/mods.conf as `load_mod_<modname> = true`
```

> * Note: Depending on the remote server's settings your client-side mods might not

Reviewer: Same as above

```suggestion
* Note: Depending on the remote server's settings, client-side mods might not
```

> * Note: Depending on the remote server's settings, client-side mods might not

Reviewer: Is the `*` at the beginning of the line required?

Author: i don't know for 2) if server can reject a mod based on name if not then yeah * is not relevant.

Reviewer: Ah, I just realised that you added the asterisk as a footnote indicator. But I don't think this is required, and could potentially add confusion (like it did for me until a couple of seconds ago). Also, our documentation files follow markdown format (in spite of having `.txt` extensions), and a `*` at the beginning of the line would render as a bullet point. It's best to avoid stray asterisks in general - you'd never know how they'd be interpreted and rendered. ;)
